### ISSUE TITLE
Snapshot upload: Improved UX

### ISSUE THREAD

**Issue by n0rdlicht** [cividi/spatial-data-package-platform#13]
- [x] While uploading a new snapshot the placeholder text in the municipality field is not showing.
- [x] Mandatory fields can be left empty before submitting a snapshot with no error message/error handling
- [x] After uploading the snapshot is shown in the map view, but not added to the snapshot list on the left, requiring a manual reload resulting in loosing log in status
- [x] adding a `?`/help button next the file upload or below the headline –  linking to project documentation (Github Wiki Pages, gemeindescan.ch/docs, cividi handbook or otherwise) explaining https://github.com/cividi/spatial-data-package-spec
- ~~validating JSON file against a schema before upload – see https://github.com/cividi/spatial-data-package-spec/issues/6~~ pushed to 0.8.0
- [x] report backend errors (REST/GraphQL) during upload or saving
- [x] Left panel overlaps modal popups

**Comment by n0rdlicht** [comment]
Left: initial modal dialog, Right: after entering and exiting the Municipality field.

<img width="925" alt="Screenshot 2021-01-14 at 11 16 21" src="https://user-images.githubusercontent.com/1425577/104578682-4edc3880-565b-11eb-87af-cfd093576cfd.png">

**Comment by n0rdlicht** [comment]
Started trying fix this by changing the `placeholder` prop to be a `label` prop. However, in addition the field is not empty on load and the municipality for editing an existing snapshot is missing.

**Comment by n0rdlicht** [comment]
@apttx the added ? looks good. When testing that locally the hit area for the button and the file field seemed to overlap with the upload winning in most cases in Safari.

**Comment by apttx** [comment]
@n0rdlicht god i hate safari. i used the v-file-input api for this, i guess i'll change it.

**Comment by apttx** [comment]
> the hit area for the button and the file field seemed to overlap with the upload winning in most cases in Safari.

fixed in #79

**Comment by apttx** [comment]
psa: hide your overflow when setting elements to be 0 wide

**Comment by apttx** [comment]
@n0rdlicht i don't know how to fix the list update, apollo query is giving me old data. graphiql shows the latest data correctly.

**Comment by n0rdlicht** [comment]
@apttx regarding the json schema. The currently valid schema is located here: https://github.com/cividi/spatial-data-package-spec/blob/main/schemas/snapshot.json.

**Comment by apttx** [comment]
@ratron ping me if you get conflicts between my prs (probably going to happen)

**Comment by apttx** [comment]
@n0rdlicht how would you like me to include the spec in this project? copy paste, clone it into this project or an npm package?

**Comment by n0rdlicht** [comment]
@apttx good question. As the map component is tied to a version of the spec there could be a "hard link" aka as a submodule or npm package to make future updates easier. What would it take to turn it into an npm package (in the spec repo?) and would that make more sense than loading it seperatly you think?

**Comment by apttx** [comment]
@n0rdlicht for maintainability an npm package is always nice, especially since you get versioning. but it's a bit more effort than just using a submodule, because every change has to be published. that being said, `npm publish` is super easy to use and can be guarded with 2fa.
you need an npm account to publish. can also be [automated](https://github.com/marketplace/actions/publish-to-npm).

to summarise:
method | pro | con
--- | --- | ---
npm | strict versioning, package is open source and easy to use, cleanly separated | has to be published, has to be installed (docker image update?), might need a build setup
submodule | easy, no additional knowledge necessary | tightly tied to this repo, no out of the box versioning

i think it makes sense for an open source specifications to be an npm package

**Comment by apttx** [comment]
that being said, publishing to npm makes the most sense if you intend others to use your package. otherwise the effort might not be worth it.

[this guide](https://zellwk.com/blog/publish-to-npm/) seems pretty good

**Comment by n0rdlicht** [comment]
> that being said, publishing to npm makes the most sense if you intend others to use your package. otherwise the effort might not be worth it.
> 
> [this guide](https://zellwk.com/blog/publish-to-npm/) seems pretty good, to give you an overview

Thanks, I would agree that we should publish the spec as an npm package.

@loleg you started thinking a bit about the spec overall [here](https://github.com/cividi/spatial-data-package-spec/pull/9#issuecomment-802632077). I could get this started and in the process address some of your comments.

**Comment by apttx** [comment]
> @apttx regarding the json schema. The currently valid schema is located here: https://github.com/cividi/spatial-data-package-spec/blob/main/schemas/snapshot.json.

@n0rdlicht I can't get this to work with ajv.
```javascript
import Ajv from 'ajv';
import snapshotSchema from '../assets/snapshot-schema.json';

const validate = new Ajv().compile(snapshotSchema);
```
What am I doing wrong?

**Comment by apttx** [comment]
> @apttx regarding the json schema. The currently valid schema is located here: https://github.com/cividi/spatial-data-package-spec/blob/main/schemas/snapshot.json.

@n0rdlicht I can't get this working with neither version of ajv. Can you give me some pointers on how to validate an object against a schema?

**Comment by n0rdlicht** [comment]
Didn't try that route yet. So far used the built in JSON schema linting support in VS Code. I will give `ajv` a try and see if it behaves differently than VS Code.

**Comment by n0rdlicht** [comment]
> @n0rdlicht I can't get this working with ajv. Can you give me some pointers on how to validate an object against a schema? I don't know what I'm doing.

Didn't try that route yet. So far used the built in JSON schema linting support in VS Code. I will give `ajv` a try and see if it behaves differently than VS Code.

**Comment by n0rdlicht** [comment]
We will push JSON validation to a later release and finalize 0.7.1 without. 

@apttx and @glaand one thing I noticed when using the current `main` is that the left panel overlaps modal popups. In my limited testing a `z-index` of `1000` was enough for above's layering issues, while `1001` is causing issues with overlays.

**Comment by n0rdlicht** [comment]
Fixed with #97
